test/rules_test.py

```diff
@@ -2,169 +2,274 @@
+    assert not evaluate_condition(
+        Condition(operator=OperatorType.GTE, value=18, attribute="age"),
+        {"age": 17},
+    )
 
 
-def test_one_of_operator_with_boolean():
-    oneOfRule = Rule(
-        allocation_key="allocation",
-        conditions=[
-            Condition(operator=OperatorType.ONE_OF, value=["True"], attribute="enabled")
-        ],
+def test_evaluate_condition_gt():
+    assert evaluate_condition(
+        Condition(operator=OperatorType.GT, value=18, attribute="age"),
+        {"age": 19},
     )
-    notOneOfRule = Rule(
-        allocation_key="allocation",
-        conditions=[
-            Condition(
-                operator=OperatorType.NOT_ONE_OF, value=["True"], attribute="enabled"
-            )
-        ],
+    assert not evaluate_condition(
+        Condition(operator=OperatorType.GT, value=18, attribute="age"),
+        {"age": 18},
     )
-    assert find_matching_rule({"enabled": True}, [oneOfRule]) == oneOfRule
-    assert find_matching_rule({"enabled": False}, [oneOfRule]) is None
-    assert find_matching_rule({"enabled": True}, [notOneOfRule]) is None
-    assert find_matching_rule({"enabled": False}, [notOneOfRule]) == notOneOfRule
 
 
-def test_one_of_operator_case_insensitive():
-    oneOfRule = Rule(
-        allocation_key="allocation",
-        conditions=[
-            Condition(
-                operator=OperatorType.ONE_OF, value=["1Ab", "Ron"], attribute="name"
-            )
-        ],
+def test_evaluate_condition_lte():
+    assert evaluate_condition(
+        Condition(operator=OperatorType.LTE, value=18, attribute="age"),
+        {"age": 18},
+    )
+    assert not evaluate_condition(
+        Condition(operator=OperatorType.LTE, value=18, attribute="age"),
+        {"age": 19},
     )
-    assert find_matching_rule({"name": "ron"}, [oneOfRule]) == oneOfRule
-    assert find_matching_rule({"name": "1AB"}, [oneOfRule]) == oneOfRule
 
 
-def test_not_one_of_operator_case_insensitive():
-    notOneOf = Rule(
-        allocation_key="allocation",
-        conditions=[
-            Condition(
-                operator=OperatorType.NOT_ONE_OF,
-                value=["bbB", "1.1.ab"],
-                attribute="name",
-            )
-        ],
+def test_evaluate_condition_lt():
+    assert evaluate_condition(
+        Condition(operator=OperatorType.LT, value=18, attribute="age"),
+        {"age": 17},
+    )
+    assert not evaluate_condition(
+        Condition(operator=OperatorType.LT, value=18, attribute="age"),
+        {"age": 18},
     )
-    assert find_matching_rule({"name": "BBB"}, [notOneOf]) is None
-    assert find_matching_rule({"name": "1.1.AB"}, [notOneOf]) is None
 
 
-def test_one_of_operator_with_string():
-    oneOfRule = Rule(
-        allocation_key="allocation",
-        conditions=[
-            Condition(
-                operator=OperatorType.ONE_OF, value=["john", "ron"], attribute="name"
-            )
-        ],
+def test_evaluate_condition_semver():
```

Comment:
semver isn't special - could be included in with the other `lt/gt` tests.

consider adding more complexity to the evaluation, such as `1.2.0 >= 1.10.5` to make sure we are comparing a semver not a string.

Reply:
Will do -- note that more complex tests are also included in the sdk-test-data